blueprints/enhanced-physlocations.md

```diff
@@ -0,0 +1,352 @@
+
+#### Data Model / Database Impact
+<!--
+*How* will this impact the Traffic Ops data model?
+*How* will this impact the Traffic Ops database schema?
+
+What changes to the lib/go-tc structs will be required?
+What new tables and columns will be required?
+How will existing tables and columns be changed?
+What are the column data types and modifiers?
+What are the FK references and constraints?
+-->
+
+Firstly the phys_location table will be altered. Simply dropping many of to be duplicated columns.
+
+
+```
+                        Table "traffic_ops.phys_location"
+      Column      |           Type           | Collation | Nullable | Default
+------------------+--------------------------+-----------+----------+---------
+id                | bigint                   |           | NOT NULL | 
+name              | text                     |           | NOT NULL |
+short_name        | text                     |           | NOT NULL | 
+region            | bigint                   |           | NOT NULL | 
+last_updated      | timestamp with time zone |           | NOT NULL | now()
+
+Indexes:    
+	idx_89655_primary PRIMARY KEY (id)
+	idx_89655_fk_phys_location_region_idx ON phys_location USING btree (region)
+	idx_89655_name_unique ON phys_location USING btree (name);
+	idx_89655_short_name_unique ON phys_location USING btree (short_name);
+Foreign-key constraints:
+	fk_phys_location_region FOREIGN KEY (region) REFERENCES region(id);
+	
+```
+
+Specifically the following columns have been dropped.
+
+* address
+* city    
+* state 
+* zip 
+* poc 
+* phone 
+* email
+* comments
+
+The first new table stores sets of addresses. This adds 3 extra address lines and changes the terminology for (state -> adminstrativeArea, city -> locality, zip -> postalCode), this is more friendly to international addresses. The other notable change is the addition of the type column. This is to be used for differentiating address. All locations in the phy_location table should have at least one full address of type `primary`, others could include a hardware mailing address or a contact address
+
+```
+                        Table "traffic_ops.location_addresses"
+      Column      |           Type           | Collation | Nullable | Default
+------------------+--------------------------+-----------+----------+---------
+id                | bigint                   |           | NOT NULL | 
+addressLine1      | text					 |           | NOT NULL | 
+addressLine2      | text					 |           | NOT NULL | 
+addressLine3      | text					 |           | NOT NULL | 
+addressLine4      | text					 |           | NOT NULL | 
+country           | text					 |           | NOT NULL | 
+administrativeArea| text					 |           | NOT NULL | 
+longitude         | bigfloat                 |           | NOT NULL | 
+latitude          | bigfloat                 |           | NOT NULL | 
```

Comment:
Why not let these be null? I don't think I've ever known the latitude and longitude for anywhere I've sent a letter.

Reply:
I am on the fence about this. It is pretty easy to find latitude and longitude information for a location. And it minimizes ambiguity potential in addresses, as well as could be useful for visualizing locations. Those are all "nice to have" but maybe not something that should be enforced.